addons/account/models/account_move.py

```diff
@@ -698,6 +698,14 @@ def default_get(self, fields):
             rec.update({'credit': balance})
         return rec
 
+    @api.multi
+    @api.constrains('credit', 'debit', 'tax_ids', 'tax_line_id', 'date')
+    def _check_tax_lock_date(self):
+        for line in self:
+            if len(line.tax_ids) and line.date <= (line.company_id.tax_lock_date or date.min):
```

Comment:
if (len(line.tax_ids) or line.tax_line_id) and line.date <= (line.company_id.tax_lock_date or date.min):